server_utils/director_discovery.go

```diff
@@ -72,18 +72,24 @@ func doDiscovery(ctx context.Context, isDirector bool) (endpoints []server_struc
 		} else {
 			log.Debugln("Server will advertise to itself using the configured advertise URL", adUrl)
 		}
-
-		endpointsTemp[adUrl] = true
-		// Bootstrap my own ad, even if nothing else is discovered.
-		if servers := directorEndpoints.Load(); servers == nil {
-			servers := make([]server_structs.DirectorAd, 1)
-			if name, err := GetServiceName(ctx, server_structs.DirectorType); err == nil {
-				servers[0] = server_structs.DirectorAd{
-					AdvertiseUrl: adUrl,
+		endpoint, err := url.Parse(adUrl)
+		if err == nil && endpoint.Port() != "0" {
+			endpointsTemp[adUrl] = true
+			// Bootstrap my own ad, even if nothing else is discovered.
+			if servers := directorEndpoints.Load(); servers == nil {
+				servers := make([]server_structs.DirectorAd, 1)
+				if name, err := GetServiceName(ctx, server_structs.DirectorType); err == nil {
+					servers[0] = server_structs.DirectorAd{
+						AdvertiseUrl: adUrl,
+					}
+					servers[0].Initialize(name)
+					directorEndpoints.CompareAndSwap(nil, &servers)
 				}
-				servers[0].Initialize(name)
-				directorEndpoints.CompareAndSwap(nil, &servers)
 			}
+		} else if err != nil {
+			log.Errorln("Ignoring URL", adUrl, "specified in Director.AdvertiseUrl due to parsing error:", err)
```

Comment:
Can you use the `param...GetName()` to set the param names in this and the next log message?